kotest-core/src/commonMain/kotlin/io/kotest/core/runtime/callbacks.kt

```diff
@@ -13,22 +13,41 @@ import io.kotest.core.test.TestResult
 import io.kotest.fp.Try
 import io.kotest.mpp.log
 
+
+fun List<ProjectListener>.resolveName() =
+   groupBy { it.name }
+      .flatMap { entry -> if (entry.value.size > 1) {
+         entry.value.mapIndexed { index, listener -> "${listener.name}_$index" to listener }
+      } else{
+         entry.value.map { it.name to it }
+      }
+      }
+
 /**
  * Invokes any afterProject functions from the given listeners.
  */
-fun List<Listener>.afterProject(): Try<Unit> = Try {
-   log("invokeAfterAll")
-   filterIsInstance<ProjectListener>().forEach { it.afterProject() }
-}.mapFailure { AfterProjectListenerException(it) }
+fun List<Listener>.afterProject(): Try<List<RuntimeException>> = Try {
+   log("invokeAfterProject")
+   filterIsInstance<ProjectListener>()
+      .resolveName()
+      .map { it.first to Try { it.second.afterProject() } }
+      .filter { it.second.isFailure() }
```

Comment:
Please add a method to Try.kt called collectFailures, which does the filter and map for you in a more elegant way.

`fun List<Try<A>>.collectFailures()`

Reply:
To be honest I don't know how to do it.
If you could do refactoring it it would be nice